### PR TITLE
Clean up federation event auth code

### PR DESCRIPTION
**Pull request by richvdh**
This is part of my work trying to sort out #9595. As a first step, this should be an (almost) non-functional change to try to make these code paths a little more understandable.

Reviewable commit by commit.

### CODE REVIEW

File: synapse/handlers/federation.py [hunk 1]
```diff
@@ -2611,16 +2611,6 @@ async def _check_event_auth(
...
 
-        # This is a hack to fix some old rooms where the initial join event
-        # didn't reference the create event in its auth events.
-        if event.type == EventTypes.Member and not event.auth_event_ids():
-            if len(event.prev_event_ids()) == 1 and event.depth < 5:
-                c = await self.store.get_event(
-                    event.prev_event_ids()[0], allow_none=True
-                )
-                if c and c.type == EventTypes.Create:
-                    auth_events[(c.type, c.state_key)] = c
```
Comment (erikjohnston) [inline synapse/handlers/federation.py:2622]: What's the reasoning for not needing this anymore? Such old rooms may still be floating around? Possibly an alternative is to only do this for v1 rooms?

**Review by erikjohnston — COMMENTED** [review]
Otherwise LGTM!

File: synapse/handlers/federation.py (hunk 1 above)
Comment (richvdh) [reply synapse/handlers/federation.py:2622]: This workaround was added in https://github.com/matrix-org/synapse/commit/858e87ab0d1bd88fcd5c819c482a40130e571874, from which I infer it was fixed before then. This is only a week after ae9c2ab1652511b36dc594882e88152b6067c816, which we know was a critical security fix. I'm pretty sure we previously checked that there weren't any remaining rooms affected by the SYN-149 bug, but if there are any other rooms that predate that fix... I don't think I want to join them.

Comment (erikjohnston) [reply synapse/handlers/federation.py:2622]: Hmm, OK fair

**Review by erikjohnston — APPROVED** [review]

**Pull request merged**